### PR TITLE
[list] Fix list_slice bug

### PR DESCRIPTION
**Pull request by maiadegraaf**
In `list_slice` when a step was used and a selection vector was applied to a child vector the size was not properly altered.  This has now been resolved.  

The `DEBUG` section has been added so this can be tested using the serializer.

### CODE REVIEW

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! LGTM. Perhaps we could add the serialization verification to `DataChunk::Verify` instead of only to the list slice code?

File: src/common/types/data_chunk.cpp [hunk 1]
```diff
@@ -328,6 +332,13 @@ void DataChunk::Verify() {
 	for (idx_t i = 0; i < ColumnCount(); i++) {
 		data[i].Verify(size());
 	}
+
+	MemoryStream mem_stream;
+	BinarySerializer serializer(mem_stream);
+
+	serializer.Begin();
+	Serialize(serializer);
+	serializer.End();
```
Comment (taniabogatsch) [inline src/common/types/data_chunk.cpp:341]: Do we want to round-trip here?
```cpp
	BinaryDeserializer deserializer(mem_stream);
	DataChunk new_chunk;

	deserializer.Begin();
	new_chunk.Deserialize(deserializer);
	deserializer.End();

	D_ASSERT(size() == new_chunk.size());
```

**Review by taniabogatsch — COMMENTED** [review]
This uncovered a few issues, hah! I added one more comment, but otherwise, this looks good; let's see if the CI finds more issues.

**Comment by taniabogatsch** [comment]
I want to look a bit more into this, we might need to merge another fix.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**